### PR TITLE
chore: make apis nullable

### PR DESCRIPTION
**Pull request by riqwan**
what:

- makes top level attributes of each object an optional field in the http layer where possible

RESOLVES CORE-2229

### CODE REVIEW

File: packages/medusa/src/api/admin/fulfillments/validators.ts [hunk 1]
```diff
@@ -32,13 +32,13 @@ export const AdminCreateFulfillment = z.object({
...
-  delivered_at: z.coerce.date().optional().nullable(),
-  canceled_at: z.coerce.date().optional().nullable(),
-  metadata: z.record(z.unknown()).optional().nullable(),
+  shipping_option_id: z.string().nullish(),
+  data: z.record(z.unknown()).nullable(),
+  packed_at: z.coerce.date().nullish(),
+  shipped_at: z.coerce.date().nullish(),
+  delivered_at: z.coerce.date().nullish(),
+  canceled_at: z.coerce.date().nullish(),
+  metadata: z.record(z.unknown()).nullable(),
```
Comment (olivermrbl) [inline packages/medusa/src/api/admin/fulfillments/validators.ts:41]: why are soem of these not `nullish`? `e.g. `metadata` is not required

**Review by olivermrbl — APPROVED** [review]
Hard to know if we have covered all by looking at the PR, but it is definitely an improvement. Let's get this merged and we can address the rest of the missing or incorrect configurations on the fly.

File: packages/medusa/src/api/admin/fulfillments/validators.ts (hunk 1 above)
Comment (riqwan) [reply packages/medusa/src/api/admin/fulfillments/validators.ts:41]: Good catch, replace all screw up. Some modules are setup for metadata to not be nullable. Shall we make them all `metadata: z.record(z.unknown()).nullish()`?

Comment (olivermrbl) [reply packages/medusa/src/api/admin/fulfillments/validators.ts:41]: I would say so yeah

File: packages/medusa/src/api/admin/campaigns/validators.ts [hunk 2]
```diff
@@ -61,10 +61,10 @@ export const AdminCreateCampaign = z
   .object({
     name: z.string(),
     campaign_identifier: z.string(),
-    description: z.string().optional(),
```
Comment (sradevski) [inline packages/medusa/src/api/admin/campaigns/validators.ts:64]: I don't think we need to explicitly set fields to `null` on create, leaving them undefined is better IMO. This also applies to the typings. WDYT?

Comment (riqwan) [reply packages/medusa/src/api/admin/campaigns/validators.ts:64]: You don't have to set this to null explicitly, it accepts undefined as well. It just doesn't throw an error if you do send null

**Pull request merged**